### ISSUE TITLE
Issues bootstrapping Juju into microk8s-on-AWS

### ISSUE THREAD

**Issue by knkski** [ubuntu/microk8s#713]
I spun up an Ubuntu server, then ran `sudo snap install microk8s --classic`. I then ran `microk8s.enable dns ingress storage`, copied over the output from `microk8s.config`, and tried to import microk8s into Juju:

    juju add-k8s uk8saws --controller uk8saws --region=aws/us-east-1 --storage juju-operator-storage

This failed first off due to security groups, and I had to run `juju run --unit ubuntu/0 "open-port 16443"` to open up the secure server port. I also ran `juju expose ubuntu`, but I'm not sure if that was also necessary in addition to the `open-port` command.

Next up was the issue of `microk8s.config` pointing at the internal IP address, and I had to manually edit the kubeconfig to point at the external IP address that I copied from the AWS console.

Finally, I had to manually edit `/var/snap/microk8s/current/certs/csr.conf.template` to add `IP.3 = $PUBLIC_IP`, and ran `sudo snap restart microk8s`. I still wasn't able to get a connection, and looking at `/var/snap/microk8s/current/certs/csr.conf.rendered` showed that that line was actually getting overwritten, and I had to put `IP.7 = $PUBLIC_IP`, restart microk8s again, and then I was able to successfully import microk8s into juju.

Not sure if this is all just stuff that should have updated documentation, or if any of it should result in code changes, but opening this issue in case anyone else runs into the same issue.

**Comment by knkski** [comment]
The above comment is oriented towards running kubectl commands locally against microk8s-on-aws. To just access the kubeflow dashboard from my laptop, I ran these commands, after installing microk8s on an AWS instance:

```
# On my laptop
juju status # Grab the public IP of the VM
juju run --unit ubuntu/0 "open-port 443"
juju expose ubuntu

# On the VM
microk8s.enable kubeflow
microk8s.kubectl edit -n kubeflow ingress/ambassador # Set spec.rules[0].host to the public IP of the VM, with `xip.io` appended, e.g. `1.2.3.4.xip.io`

# Then go to the xip.io address on my laptop and log in
```

**Comment by ktsakalozos** [comment]
@knkski that's some cool information on setting up kubeflow+microk8s+aws. Should we put it down on a tutorial/site? @evilnick WDYT?

**Comment by sakaia** [comment]
@knkski thank you for your information.
Is your setting AWS::EC2::SecurityGroupIngress is open to port 80 in default?

```yaml
spec:
  rules:
  - host: xxx.xxx.xxx.xxx.xip.io
    http:
      paths:
      - backend:
          serviceName: ambassador
          servicePort: 80
        path: /
        pathType: ImplementationSpecific
```
In my setting, following messages are appeared and cannot access xxx.xxx.xxx.xxx.xip.io.

```
I0408 00:41:47.457123       6 event.go:258] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"kubeflow", Name:"ambassador", UID:"0bfb08b1-746d-4314-a2c1-755ca25d869c", APIVersion:"networking.k8s.io/v1beta1", ResourceVersion:"95831", FieldPath:""}): type: 'Normal' reason: 'UPDATE' Ingress kubeflow/ambassador
W0408 00:41:50.299770       6 controller.go:878] Service "kubeflow/ambassador" does not have any active Endpoint.
I0408 00:41:50.299858       6 controller.go:133] Configuration changes detected, backend reload required.
I0408 00:41:50.973174       6 controller.go:149] Backend successfully reloaded.
W0408 00:41:53.633061       6 controller.go:878] Service "kubeflow/ambassador" does not have any active Endpoint.
W0408 00:41:56.966361       6 controller.go:878] Service "kubeflow/ambassador" does not have any active Endpoint.
W0408 00:42:00.300179       6 controller.go:878] Service "kubeflow/ambassador" does not have any active Endpoint.
```

**Comment by sakaia** [comment]
@knkski 

Port 443 opens for AWS EC2 then the packet reaches ingress. But Still have following error.
Is there any advise to solve this?
I ran this on microk8s 1.18.0.

```
$ kubectl logs -n ingress nginx-ingress-microk8s-controller-xxxxx
```

```bash
yyy.yyy.yyy.yyy - [yyy.yyy.yyy.yyy] - - [09/Apr/2020:04:22:05 +0000] "GET / HTTP/1.1" 400 663 "-" "Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.132 Safari/537.36" 461 0.000 [] [] - - - - 7f9c1b247d9c4a10ca78dfcfb389ddea
yyy.yyy.yyy.yyy - [yyy.yyy.yyy.yyy] - - [09/Apr/2020:04:22:07 +0000] "GET /favicon.ico HTTP/1.1" 400 663 "http://zzz.zzz.zzz.zzz:443/" "Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.132 Safari/537.36" 393 0.000 [] [] - - - - f9ba190f98523d4146a7573a289803c8
yyy.yyy.yyy.yyy - [yyy.yyy.yyy.yyy] - - [09/Apr/2020:04:22:41 +0000] "GET /favicon.ico HTTP/1.1" 400 663 "http://zzz.zzz.zzz.zzz:443/" "Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.132 Safari/537.36" 393 0.000 [] [] - - - - a6bc0a7d43f15fffac8311eec45a5e22
```

**Comment by knkski** [comment]
@sakaia: Can you post the output from `microk8s.kubectl logs -n kubeflow --tail 1000 -l juju-app=ambassador`?

**Comment by sakaia** [comment]
Thank you for comments. 

I attach the file
[Uploading 1130.log…](1130.log)

I wanto to confirm. Because I set IP address limit to local pc only, so Can I connect directry to AWS EC2 Port 80?
(I guess this case it cannot use xip.io service for this case )


```
 kubectl -n kubeflow describe service/ambassador
Name:                     ambassador
Namespace:                kubeflow
Labels:                   juju-app=ambassador
Annotations:              juju.io/controller: f2279aba-7a80-49f0-8ec1-ad50e47f9808
                          juju.io/model: 1da2d170-29f0-4b45-8e1f-6ab43353b291
                          metallb.universe.tf/address-pool: default
Selector:                 juju-app=ambassador
Type:                     LoadBalancer
IP:                       10.152.183.174
LoadBalancer Ingress:     10.64.140.43
Port:                     ambassador  80/TCP
TargetPort:               80/TCP
NodePort:                 ambassador  30932/TCP
Endpoints:                10.1.45.154:80
Session Affinity:         None
External Traffic Policy:  Cluster
Events:
  Type    Reason        Age                  From             Message
  ----    ------        ----                 ----             -------
  Normal  nodeAssigned  73s (x24 over 155m)  metallb-speaker  announcing from node "ip-10-10-174-153"
```

**Comment by paravatha** [comment]
@knkski , @sakaia 
 
Still not working after trying these. But, I do see my requests in the ambassador and ingress
**1. I updated the host with my public ip using microk8s.kubectl edit -n kubeflow ingress/ambassador** 

**2. microk8s.kubectl logs -n kubeflow --tail 100 -l juju-app=ambassador**

```
ACCESS [2020-04-11T21:56:59.856Z] "GET /dex/.well-known/openid-configuration HTTP/1.1" 200 - 0 649 0 0 "10.1.40.1" "Go-http-client/1.1" "424e0aeb-4d9d-4ef6-a64f-1c3a9686c0ea" "10.64.140.44.xip.io:80" "10.152.183.136:5556"
ACCESS [2020-04-11T21:57:09.863Z] "GET /dex/.well-known/openid-configuration HTTP/1.1" 200 - 0 649 0 0 "10.1.40.1" "Go-http-client/1.1" "7b59be4a-f5e1-4dfe-a448-d035969e40cc" "10.64.140.44.xip.io:80" "10.152.183.136:5556"
ACCESS [2020-04-11T21:57:19.865Z] "GET /dex/.well-known/openid-configuration HTTP/1.1" 200 - 0 794 1 0 "10.1.40.1" "Go-http-client/1.1" "1909e7b5-a5a6-4a88-a1b8-c1070969c456" "10.64.140.44.xip.io:80" "10.152.183.136:5556"
ACCESS [2020-04-11T22:28:04.764Z] "GET / HTTP/1.1" 302 UAEX 0 333 5 - "my.laptop.ip.addr,10.1.40.1" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36" "1ff7d339-828a-446c-a255-ea5e8b6e6fb0" "my.ec2.public.ip.xip.io" "-"
ACCESS [2020-04-11T22:28:47.772Z] "GET / HTTP/1.1" 302 UAEX 0 333 2 - "my.laptop.ip.addr,10.1.40.1" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36" "935b03ec-98aa-4088-ab81-842132c73d07" "my.ec2.public.ip.xip.io" "-"
ACCESS [2020-04-11T22:45:11.293Z] "GET / HTTP/1.1" 302 UAEX 0 333 2 - "my.laptop.ip.addr,10.1.40.1" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36" "72bd7e3d-cac1-4572-af31-b4879149993a" "my.ec2.public.ip.xip.io" "-"
ACCESS [2020-04-11T22:45:32.799Z] "GET / HTTP/1.1" 302 UAEX 0 333 2 - "my.laptop.ip.addr,10.1.40.1" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36" "b8a8f375-38ea-4d98-8451-6000749c614f" "my.ec2.public.ip.xip.io" "-"
```

**3. kubectl logs --tail=100  -n ingress nginx-ingress-microk8s-controller-pvlwx**

```
I0411 21:53:20.300805       6 status.go:296] updating Ingress kubeflow/ambassador status from [] to [{127.0.0.1 }]
I0411 21:53:20.339787       6 event.go:258] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"kubeflow", Name:"ambassador", UID:"ffda12ee-3309-4a02-bf28-19bc02e33011", APIVersion:"networking.k8s.io/v1beta1", ResourceVersion:"10271", FieldPath:""}): type: 'Normal' reason: 'UPDATE' Ingress kubeflow/ambassador
I0411 22:27:08.965683       6 event.go:258] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"kubeflow", Name:"ambassador", UID:"ffda12ee-3309-4a02-bf28-19bc02e33011", APIVersion:"networking.k8s.io/v1beta1", ResourceVersion:"18077", FieldPath:""}): type: 'Normal' reason: 'UPDATE' Ingress kubeflow/ambassador
I0411 22:27:08.965953       6 controller.go:133] Configuration changes detected, backend reload required.
I0411 22:27:09.046164       6 controller.go:149] Backend successfully reloaded.
my.laptop.ip.addr - [my.laptop.ip.addr] - - [11/Apr/2020:22:28:04 +0000] "GET / HTTP/1.1" 302 333 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36" 457 0.007 [kubeflow-ambassador-80] [] 10.1.40.31:80 333 0.008 302 18640a131f2e91c7cd14b82f7e0d5d0f
my.laptop.ip.addr - [my.laptop.ip.addr] - - [11/Apr/2020:22:28:47 +0000] "GET / HTTP/1.1" 302 333 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36" 457 0.003 [kubeflow-ambassador-80] [] 10.1.40.31:80 333 0.004 302 1479a9d3fdd871c782ba98e4cdb8b3ec
my.laptop.ip.addr - [my.laptop.ip.addr] - - [11/Apr/2020:22:45:11 +0000] "GET / HTTP/1.1" 302 333 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36" 457 0.004 [kubeflow-ambassador-80] [] 10.1.40.31:80 333 0.000 302 50fa335c365bb4144ddb2663154101c2
my.laptop.ip.addr - [my.laptop.ip.addr] - - [11/Apr/2020:22:45:32 +0000] "GET / HTTP/1.1" 302 333 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36" 457 0.003 [kubeflow-ambassador-80] [] 10.1.40.31:80 333 0.004 302 ca29a4996200a24b42e6fb520cbeab2a

```

**Comment by paravatha** [comment]
Wondering if its proxy issue

****sudo microk8s.kubectl proxy --accept-hosts=.* --address=0.0.0.0 &****

After running the proxy command, I am able to access k8s dashboard and Grafana  
http://my.ec2.public.ip:8001/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:/proxy
http://my.ec2.public.ip:8001/api/v1/namespaces/kube-system/services/monitoring-grafana/proxy

**Comment by sakaia** [comment]
@knkski, @paravatha 

Thank you for comment.
1)In my environment, following procedure also works fine.
But kubeflow WebUI is not appeared

> Wondering if its proxy issue
> 
> _**_sudo microk8s.kubectl proxy --accept-hosts=._ --address=0.0.0.0 &**_*
> 
> After running the proxy command, I am able to access k8s dashboard and Grafana
> http://my.ec2.public.ip:8001/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:/proxy
> http://my.ec2.public.ip:8001/api/v1/namespaces/kube-system/services/monitoring-grafana/proxy


2)From external IP, returns 404 Not Founderror by openresty/1.15.8.1
In this case nothing logged even ingress.

chrome http://externalIP


```bash
$ kubectl logs -n ingress  nginx-ingress-microk8s-controller-6254j
-------------------------------------------------------------------------------
NGINX Ingress controller
  Release:       0.25.1
  Build:         git-5179893a9
  Repository:    https://github.com/kubernetes/ingress-nginx/
  nginx version: openresty/1.15.8.1

-------------------------------------------------------------------------------

W0413 02:18:53.798533       6 flags.go:221] SSL certificate chain completion is disabled (--enable-ssl-chain-completion=false)
nginx version: openresty/1.15.8.1
W0413 02:18:53.803068       6 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0413 02:18:53.803250       6 main.go:183] Creating API client for https://10.152.183.1:443
I0413 02:18:53.814973       6 main.go:227] Running in Kubernetes cluster version v1.18 (v1.18.0) - git (clean) commit 9e991415386e4cf155a24b1da15becaa390438d8 - platform linux/amd64
I0413 02:18:53.934081       6 main.go:102] Created fake certificate with PemFileName: /etc/ingress-controller/ssl/default-fake-certificate.pem
I0413 02:18:53.968316       6 nginx.go:274] Starting NGINX Ingress controller
I0413 02:18:54.045726       6 event.go:258] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"ingress", Name:"nginx-load-balancer-microk8s-conf", UID:"4e55bf90-cc5f-4f88-a723-1b0293563e04", APIVersion:"v1", ResourceVersion:"503", FieldPath:""}): type: 'Normal' reason: 'CREATE' ConfigMap ingress/nginx-load-balancer-microk8s-conf
I0413 02:18:55.168995       6 nginx.go:318] Starting NGINX process
I0413 02:18:55.169224       6 leaderelection.go:235] attempting to acquire leader lease  ingress/ingress-controller-leader-nginx...
I0413 02:18:55.170369       6 controller.go:133] Configuration changes detected, backend reload required.
I0413 02:18:55.180423       6 leaderelection.go:245] successfully acquired lease ingress/ingress-controller-leader-nginx
I0413 02:18:55.181289       6 status.go:86] new leader elected: nginx-ingress-microk8s-controller-6254j
I0413 02:18:55.268179       6 controller.go:149] Backend successfully reloaded.
I0413 02:18:55.268227       6 controller.go:158] Initial sync, sleeping for 1 second.
E0413 02:20:18.804420       6 leaderelection.go:324] error retrieving resource lock ingress/ingress-controller-leader-nginx: etcdserver: request timed out
E0413 02:20:25.526486       6 event.go:296] Could not construct reference to: '&v1.ConfigMap{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Data:map[string]string(nil), BinaryData:map[string][]uint8(nil)}' due to: 'selfLink was empty, can't make reference'. Will not report event: 'Normal' 'LeaderElection' 'nginx-ingress-microk8s-controller-6254j stopped leading'
I0413 02:20:25.526587       6 leaderelection.go:281] failed to renew lease ingress/ingress-controller-leader-nginx: timed out waiting for the condition
I0413 02:20:25.526642       6 leaderelection.go:235] attempting to acquire leader lease  ingress/ingress-controller-leader-nginx...
I0413 02:20:25.706328       6 leaderelection.go:245] successfully acquired lease ingress/ingress-controller-leader-nginx
```


3)After doing curl 2times from login console, it logs to the ambassador but returns 403 code
```bash
$ curl --verbose http://10.64.140.43
```
ambassador

```bash
2020-04-13 02:28:01 diagd 0.50.0 [P65TAmbassadorEventWatcher] INFO: Scout notices: [{"level": "INFO", "message": "Upgrade available! to Ambassador version 1.4.0"}]
2020-04-13 02:28:01 diagd 0.50.0 [P65TAmbassadorEventWatcher] INFO: App notices after scout: [{"level": "INFO", "message": "Upgrade available! to Ambassador version 1.4.0"}]
2020-04-13 02:28:01 diagd 0.50.0 [P65TAmbassadorEventWatcher] INFO: extauth: server_uri http://
2020-04-13 02:28:01 diagd 0.50.0 [P65TAmbassadorEventWatcher] INFO: saving Envoy configuration for snapshot 60
2020-04-13 02:28:01 diagd 0.50.0 [P65TAmbassadorEventWatcher] INFO: notifying PID 29 ambex
2020-04-13 02:28:01 diagd 0.50.0 [P65TAmbassadorEventWatcher] INFO: configuration updated
time="2020-04-13T02:28:01Z" level=info msg="Loaded file /ambassador/envoy/envoy.json"
time="2020-04-13T02:28:01Z" level=info msg="Pushing snapshot v60"
ACCESS [2020-04-13T02:28:23.759Z] "GET / HTTP/1.1" 403 UAEX 0 0 10 - "10.1.45.1" "curl/7.65.3" "1d363df6-6fc1-47e4-a845-9ebb0a48fe9f" "10.64.140.43" "-"
ACCESS [2020-04-13T02:28:36.797Z] "GET / HTTP/1.1" 403 UAEX 0 0 0 - "10.1.45.1" "curl/7.65.3" "a518ab14-e838-4fe9-8a73-3b7242d1a6d4" "10.64.140.43" "-"
[2020-04-13 02:36:35.316][000030][info][main] [source/server/drain_manager_impl.cc:63] shutting down parent after drain
```

**Comment by sakaia** [comment]
> @knkski , @sakaia
> 
> Still not working after trying these. But, I do see my requests in the ambassador and ingress

> **2. microk8s.kubectl logs -n kubeflow --tail 100 -l juju-app=ambassador**
> 
> ```
> ACCESS [2020-04-11T21:56:59.856Z] "GET /dex/.well-known/openid-configuration HTTP/1.1" 200 - 0 649 0 0 "10.1.40.1" "Go-http-client/1.1" "424e0aeb-4d9d-4ef6-a64f-1c3a9686c0ea" "10.64.140.44.xip.io:80" "10.152.183.136:5556"
> ACCESS [2020-04-11T21:57:09.863Z] "GET /dex/.well-known/openid-configuration HTTP/1.1" 200 - 0 649 0 0 "10.1.40.1" "Go-http-client/1.1" "7b59be4a-f5e1-4dfe-a448-d035969e40cc" "10.64.140.44.xip.io:80" "10.152.183.136:5556"
> ACCESS [2020-04-11T21:57:19.865Z] "GET /dex/.well-known/openid-configuration HTTP/1.1" 200 - 0 794 1 0 "10.1.40.1" "Go-http-client/1.1" "1909e7b5-a5a6-4a88-a1b8-c1070969c456" "10.64.140.44.xip.io:80" "10.152.183.136:5556"
> ACCESS [2020-04-11T22:28:04.764Z] "GET / HTTP/1.1" 302 UAEX 0 333 5 - "my.laptop.ip.addr,10.1.40.1" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36" "1ff7d339-828a-446c-a255-ea5e8b6e6fb0" "my.ec2.public.ip.xip.io" "-"
> ACCESS [2020-04-11T22:28:47.772Z] "GET / HTTP/1.1" 302 UAEX 0 333 2 - "my.laptop.ip.addr,10.1.40.1" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36" "935b03ec-98aa-4088-ab81-842132c73d07" "my.ec2.public.ip.xip.io" "-"
> ACCESS [2020-04-11T22:45:11.293Z] "GET / HTTP/1.1" 302 UAEX 0 333 2 - "my.laptop.ip.addr,10.1.40.1" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36" "72bd7e3d-cac1-4572-af31-b4879149993a" "my.ec2.public.ip.xip.io" "-"
> ACCESS [2020-04-11T22:45:32.799Z] "GET / HTTP/1.1" 302 UAEX 0 333 2 - "my.laptop.ip.addr,10.1.40.1" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36" "b8a8f375-38ea-4d98-8451-6000749c614f" "my.ec2.public.ip.xip.io" "-"
> ```

It seems `UAEX` problem. I think default installation misses something.

Ref
https://blog.getambassador.io/understanding-envoy-proxy-and-ambassador-http-access-logs-fee7802a2ec5


p.s. My Installation procedure is simple on 2core and 32GB on AWS EC2.

```bash
$ sudo snap install microk8s --classic --channel=1.18/stable
$ sudo usermod -a -G microk8s ubuntu
$ sudo chown -f -R ubuntu ~/.kube
```
relogin

```bash
$ sudo snap alias microk8s.kubectl kubectl
$ microk8s.status --wait-ready
$ microk8s.enable dns dashboard storage
$ microk8s.enable kubeflow
$ $ microk8s.kubectl edit -n kubeflow ingress/ambassador # address fixation
```

**Comment by sakaia** [comment]
From seeing metallb document, it seems it does not work for EC2 with metallb (both BGP/L2 mode). Is my understanding correct?

Why Doesn’t MetalLB Work On (Most) Cloud Platforms?
https://metallb.universe.tf/installation/clouds/#why-doesn-t-metallb-work-on-most-cloud-platforms

Kubernetes NodePort vs LoadBalancer vs Ingress? When should I use what?
https://medium.com/google-cloud/kubernetes-nodeport-vs-loadbalancer-vs-ingress-when-should-i-use-what-922f010849e0

**Comment by paravatha** [comment]
@sakaia , Sorry for the delay in responding
I got it working partially by referring to this https://github.com/kubeflow/manifests/issues/974 
I can access the central dashboard, but not some of the under sub-menu items

**Comment by sakaia** [comment]
@paravatha thank you for notification.
I also uses `kubectl port-forward` and it seems work fine for jupyternotebook etc.

**Comment by knkski** [comment]
@paravatha: Can you file a new issue with details about the sub-menu items you're unable to access?

**Comment by meghas186** [comment]
@knkski @paravatha - I changed ip address in ingress. but still I am not able to access the dashboard outside my VM. attaching ambassador logs which is saying 302 response code - 

ACCESS [2020-05-27T17:29:30.832Z] "GET / HTTP/1.1" 302 UAEX 0 325 2 - "my.laptop.ip.address,10.1.85.1" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36" "f103a9ff-2a3e-4677-bf44-8e3ea3f079bf" "ec2.instance.ipv4.address.xip.io" "-"

Could you please help.

**Comment by paravatha** [comment]
@meghas186 , Please try these port-forward commands
https://github.com/kubeflow/manifests/issues/974#issuecomment-612546805

**Comment by paravatha** [comment]
This issue is resolved, please refer to #1140

**Comment by ktsakalozos** [comment]
Thank you all for your input. Feel free to reopen the issue if you feel it is not resolved.
